# SPDX-License-Identifier: AGPL-3.0-only
# Copyright (c) 2026 DH0516
# Use the recipe's job limit for GMP and MPFR.
--- a/deps/GMP/GMP.cmake
+++ b/deps/GMP/GMP.cmake
@@ -66,7 +66,7 @@
         PATCH_COMMAND git apply ${GMP_DIRECTORY_FLAG} --verbose ${CMAKE_CURRENT_LIST_DIR}/0001-GMP_GCC15.patch
         BUILD_IN_SOURCE ON
         CONFIGURE_COMMAND  env "CC=${CMAKE_C_COMPILER}" "CXX=${CMAKE_CXX_COMPILER}" "CFLAGS=${_gmp_ccflags}" "CXXFLAGS=${_gmp_ccflags}" "LDFLAGS=${CMAKE_EXE_LINKER_FLAGS}" ./configure ${_cross_compile_arg} --enable-shared=no --enable-cxx=yes --enable-static=yes "--prefix=${DESTDIR}" ${_gmp_build_tgt}
-        BUILD_COMMAND     make -j
+        BUILD_COMMAND     make -j$ENV{CMAKE_BUILD_PARALLEL_LEVEL}
         INSTALL_COMMAND   make install
     )
 endif ()
--- a/deps/MPFR/MPFR.cmake
+++ b/deps/MPFR/MPFR.cmake
@@ -32,7 +32,7 @@
         BUILD_IN_SOURCE ON
         CONFIGURE_COMMAND autoreconf -f -i && 
                           env "CC=${CMAKE_C_COMPILER}" "CXX=${CMAKE_CXX_COMPILER}" "CFLAGS=${_gmp_ccflags}" "CXXFLAGS=${_gmp_ccflags}" "LDFLAGS=${CMAKE_EXE_LINKER_FLAGS}" ./configure ${_cross_compile_arg} --prefix=${DESTDIR} --enable-shared=no --enable-static=yes --with-gmp=${DESTDIR} ${_gmp_build_tgt}
-        BUILD_COMMAND make -j
+        BUILD_COMMAND make -j$ENV{CMAKE_BUILD_PARALLEL_LEVEL}
         INSTALL_COMMAND make install
         DEPENDS dep_GMP
     )
